src/main/resources/db/migration/V109__ViabilityTestSelections.sql

```diff
@@ -0,0 +1,18 @@
+-- Each accession can optionally have a single viability test that is considered the "selected" one.
+-- There are three ways we could model this: as a "selected test ID" column on the accessions table,
+-- as an "is selected" flag on the viability_tests table, or as a separate table. The first two
+-- options are more complicated to update when, e.g., the selected test is being deleted or the
+-- selection is switching to a newly-inserted test.
+
+
+-- viability_tests.id is already unique, but we want to use a foreign key to guarantee that we don't
+-- mark a test from accession 1 as the selection for accession 2, and the target columns of a
+-- foreign key have to have a unique constraint.
+ALTER TABLE viability_tests ADD CONSTRAINT viability_tests_id_accession_unique UNIQUE (id, accession_id);
+
+CREATE TABLE viability_test_selections (
+    accession_id BIGINT PRIMARY KEY REFERENCES accessions,
+    viability_test_id BIGINT NOT NULL,
+    UNIQUE (viability_test_id),
```

Comment:
dumb question, possible to have a complex unique independent key list of accession_id and viability_test_id?

Reply:
Not sure I quite understand the question. You mean have `UNIQUE (accession_id, viability_test_id)` in addition to the existing primary key and `UNIQUE`? That'd definitely be possible, but it wouldn't constrain things any further.

Reply:
Unique independent of each other... not necessarily as a tuple.
ex. accession_id should be unique and same for viability_test_id
I guess avoid inserting an accession id more than once.
It was just a question, not a biggie :)

Reply:
Oops.. i missed the primary key on accession_id.